apps/studio/schemas/boardYear.ts

```diff
@@ -0,0 +1,70 @@
+import { defineField, defineType, defineArrayMember } from "sanity";
+import { HeartIcon } from "@sanity/icons";
+
+const personSearch = [
+  defineArrayMember({
+    type: "object",
+    fields: [
+      defineField({
+        name: "person",
+        type: "reference",
+        title: "Person",
+        to: {
+          type: "person",
+        },
+        validation: (Rule) => Rule.required(),
+      }),
+      defineField({
+        name: "position",
+        type: "string",
+        title: "Position",
+        validation: (Rule) => Rule.required(),
+      }),
+    ],
+    preview: {
+      select: {
+        title: "person.name",
+        subtitle: "position",
+      },
+    },
+    icon: HeartIcon,
+  }),
+];
+
+const nameTitleAssoc: Map<string, string> = new Map([
+  ["corporate", "Corporate Organizers"],
+  ["logistics", "Logistics Organizers"],
+  ["marketing", "Marketing Organizers"],
+  ["tech", "Tech Organizers"],
+]);
+
+export default defineType({
+  name: "boardYear",
+  type: "document",
+  title: "Board Years",
+  fields: [
+    defineField({
+      name: "year",
+      type: "number",
+      title: "Academic Year",
+      validation: (Rule) => Rule.required(),
+    }),
+    ...Object.entries(nameTitleAssoc).map(([key, title]) =>
+      defineField({
+        name: key,
+        type: "array",
+        title: title,
+        of: personSearch,
```

Comment:
I think it might make more sense to move the `[]` and `defineArrayMember` from above to here. That way all the syntax for the array (`type: "array"`, `of:`, etc.) is located in this area of the schema, while the object is clearly defined above.

Reply:
Makes sense! Fixed in [`be8cb05`](https://github.com/HackAtUCI/hack.ics.uci.edu/pull/12/commits/be8cb056605ba8708d4ceafed6fa6af9e446787e)